include/anyode/anyode_matrix.hpp

```diff
@@ -1,104 +1,135 @@
+
 #include <stdint.h> // uintptr_t
+#include <cstdlib>  // std::aligned_alloc (C++17) & std::free
 #include <cstring>  // std::memset
 #include <stdexcept> // std::runtime_error
 
-#if ANYODE_NO_LAPACK == 1
-#include "anyode/anyode_blasless.hpp"
-#else
+#ifndef ANYODE_USE_LAPACK
+#define ANYODE_USE_LAPACK 0
+#endif
+
+#if ANYODE_USE_LAPACK == 1
 #include "anyode/anyode_blas_lapack.hpp"
+#else
+#include "anyode/anyode_blasless.hpp"
 #endif
 
 namespace AnyODE {
-    template<typename T> constexpr std::size_t n_padded(std::size_t n, int alignment_bytes){
-        return ((n*sizeof(T) + alignment_bytes - 1) & ~(alignment_bytes - 1)) / sizeof(T);
+    template<typename T>
+    constexpr std::size_t n_padded(std::size_t n, int alignment_bytes) {
+        return ((n * sizeof(T) + alignment_bytes - 1) & ~(alignment_bytes - 1)) / sizeof(T);
     }
+
     static constexpr int alignment_bytes_ = 64; // L1 cache line
 
     template<typename Real_t>
     class MatrixBase {
-        void * m_own_array_ = nullptr;
-        Real_t * alloc_array_(int n){
-            m_own_array_ = std::malloc(sizeof(Real_t)*n + alignment_bytes_ - 1);
+        void *m_own_array_ = nullptr;
+
+        Real_t *alloc_array_(int n) {
+            m_own_array_ = std::malloc(sizeof(Real_t) * n + alignment_bytes_ - 1);
             const uintptr_t mask = ~uintptr_t(alignment_bytes_ - 1);
             const uintptr_t addr = reinterpret_cast<uintptr_t>(m_own_array_);
             const uintptr_t candidate = addr + alignment_bytes_ - 1;
             return static_cast<Real_t *>(reinterpret_cast<void *>(candidate & mask));
         }
+
     public:
 
-        Real_t * m_data;
+        Real_t *m_data;
         int m_nr, m_nc, m_ld, m_ndata;
         bool m_own_data;
-        MatrixBase(Real_t * const data, int nr, int nc, int ld, int ndata, bool own_data=false) :
-            m_data(data ? data : alloc_array_(ndata)), m_nr(nr), m_nc(nc), m_ld(ld), m_ndata(ndata),
-            m_own_data(own_data)
-        {
-            if (data == nullptr && own_data)
+
+        MatrixBase(Real_t *const data, int nr, int nc, int ld, int ndata, bool own_data = false) :
+                m_data(data ? data : alloc_array_(ndata)), m_nr(nr), m_nc(nc), m_ld(ld), m_ndata(ndata),
+                m_own_data(own_data) {
+            if (data == nullptr and own_data)
                 throw std::runtime_error("own_data not needed for nullptr");
         }
-        MatrixBase(const MatrixBase<Real_t>& ori) : MatrixBase(nullptr, ori.m_nr, ori.m_nc, ori.m_ld, ori.m_ndata) {
+
+        MatrixBase(const MatrixBase<Real_t> &ori) : MatrixBase(nullptr, ori.m_nr, ori.m_nc, ori.m_ld, ori.m_ndata) {
             std::copy(ori.m_data, ori.m_data + m_ndata, m_data);
         }
-        virtual ~MatrixBase(){
+
+        virtual ~MatrixBase() {
             if (m_own_array_)
                 std::free(m_own_array_);
-            if (m_own_data && m_data)
+            if (m_own_data and m_data)
```

Comment:
again MSVC was the reason I changed this: we'll have to live with &&, || (since they don't support `and`, `or`)